application/forms/EventRuleConfigForm.php

```diff
@@ -0,0 +1,647 @@
+                            if ($elementValue === null) {
+                                continue;
+                            }
+
+                            $count = $key + 1;
+                            $selectedOption = str_replace('id', $elementValue, $elementName, $replaced);
+                            if ($replaced && $elementName !== 'channel_id') {
+                                $recipientFormValues['column_' . $count] = $selectedOption;
+                            } elseif ($elementName === 'channel_id') {
+                                $recipientFormValues['val_' . $count] = $elementValue;
+                            }
+                        }
+
+                        if (isset($recipient['id'])) {
+                            $recipientFormValues['id_' . $count] = (int) $recipient['id'];
+                        }
+                    }
+                }
+            }
+
+            $formValues['escalation-recipient_' . bin2hex($position)] = $recipientFormValues;
+        }
+
+        return parent::populate($formValues);
+    }
+
+    /**
+     * Get the values for the current EventRuleConfigForm
+     *
+     * @return array<string, mixed> values as name-value pairs
+     */
+    public function getValues(): array
+    {
+        $values = [];
+        $escalations = [];
+        $prefixesString = $this->getValue('prefixes-map', '');
+
+        /** @var string[] $prefixesMap */
+        $prefixesMap = explode(',', $prefixesString);
+        $i = 1;
+        foreach ($prefixesMap as $prefixMap) {
+            /** @var EscalationCondition $escalationCondition */
+            $escalationCondition = $this->getElement('escalation-condition_' . $prefixMap);
+            /** @var EscalationRecipient $escalationRecipient */
+            $escalationRecipient = $this->getElement('escalation-recipient_' . $prefixMap);
+            $escalations[$i]['condition'] = $escalationCondition->getCondition();
+            $escalations[$i]['id'] = $escalationCondition->getValue('id');
+            $escalations[$i]['recipients'] = $escalationRecipient->getRecipients();
+            $i++;
+        }
+
+        /** @var EventRuleConfigFilter $configFilter */
+        $configFilter = $this->getElement('config-filter');
+        $values['object_filter'] = $configFilter->getObjectFilter();
+        $values['rule_escalation'] = $escalations;
+
+        return $values;
+    }
+
+    /**
+     *  Create remove button for the given escalation position
+     *
+     * @param string $prefix
+     *
+     * @return SubmitButtonElement
+     */
+    protected function createRemoveButton(string $prefix): SubmitButtonElement
+    {
+        /** @var array<int, array<string, mixed>> $escalations */
+        $escalations = $this->config['rule_escalation'] ?? [];
+        $pos = hex2bin($prefix);
+        $disableRemoveButton = false;
+        $escalationId = $escalations[$pos]['id'] ?? null;
+
+        if ($escalationId && ctype_digit($escalationId)) {
```

Comment:
What else could these be than digits?

Reply:
This was used because I am assigning hexadecimal values as fake IDs for escalations that are not yet saved. In this case using `is_numeric` to avoid the call to database does not work as that would be true even if the IDs are hexadecimal.